common/scoring.ts

```diff
@@ -62,7 +63,25 @@ export function scoreUsersByContract(contract: Contract, bets: Bet[]) {
       return { userId, payout }
     })
 
-  const netPayouts = [...resolvePayouts, ...salePayouts, ...investments]
+  const profits = openBets.map((bet) => {
+    const { userId } = bet
+    if (contract.isResolved) {
+      return { userId: userId, payout: 0 }
+    }
+    const payout = getContractBetMetrics(contract, [bet]).payout
+    console.log({
+      userId: userId,
+      metrics: getContractBetMetrics(contract, [bet]),
+    })
+    return { userId: userId, payout: payout }
+  })
+
+  const netPayouts = [
+    ...resolvePayouts,
+    ...salePayouts,
+    ...investments,
+    ...profits,
```

Comment:
I think the profit payouts include profits from resolution and sales and investment. So I think you can replace all this code with just the profit payouts. (The previous scoring code looks pretty old.)

For groups, I think we're going to want to not fetch all the contracts and bets to compute this in getStaticProps, so we should aim to do it in the `updateMetrics` cloud function.

Reply:
I early return in the profits method if the bets are resolved, so it is working as expected.

But fair, if we can replace all of it that's better, I'll look into it.